src/writer.js

```diff
@@ -19,11 +19,17 @@ class Writer {
   }
 
   append (span) {
-    const trace = span.context()._trace
+    const spanContext = span.context()
+    const trace = spanContext._trace
 
     if (trace.started.length === trace.finished.length) {
       const formattedTrace = trace.finished.map(format)
 
+      if (spanContext._sampling.drop === true) {
+        log.debug(() => `Dropping trace due to user configured filtering: ${JSON.stringify(formattedTrace)}`)
```

Comment:
Using `JSON.stringify` could be very verbose and prone to circular dependency issues. What is the expected output?

Reply:
Hmm its used right below on line 33 https://github.com/DataDog/dd-trace-js/pull/438/files/1172d3122e0b9d2bbe9841955aebcab921882aa9#diff-7dec243e7443d2aa55101241b38a8575R33

Reply:
Mainly the intention is to report what is being dropped for debugging purposes

Reply:
Can't argue with that 😅